xds/src/main/java/io/grpc/xds/XdsNameResolver.java

```diff
@@ -300,15 +412,74 @@ public void run() {
     }
   }
 
-  // https://github.com/google/error-prone/issues/1767
-  @SuppressWarnings("ModifyCollectionInEnhancedForLoop")
```

Comment:
I wonder where the error-prone warning was coming from in the old code.

Reply:
It's for modifying ConcurrentMap `clusterRefs` in the loop. Not sure how that was triggered. But now, seems it doesn't trigger.